lib/declaration.ts

```diff
@@ -316,8 +323,23 @@ export default class DeclarationTranspiler extends base.TranspilerBase {
     });
 
     orderedGroups.forEach((group: Array<ts.Node>) => {
+      const first = group[0] as ts.SignatureDeclaration;
+      // Check if the methods in this group return Promises. If so, skip visiting them on first
+      // encounter and add them to the promiseMethods set. If they are already in promiseMethods
```

Comment:
that is a bit fiddly.
rather than having a method that executes differently the second time you run it, explicitly change some state when you are attempting to emit promises and when you aren't.